rearrange_cars_luiza.py

```diff
@@ -0,0 +1,253 @@
+        one car not yet arranged
+        '''
+
+        index = 0
+        while self.current_car_order[index] == self.final_car_order[index]:
+            index += 1
+            if index == self.empty_spot_position:
+                index += 1
+
+        return index
+
+    def rearrange_one_car(self):
+
+        '''Puts to the current empty space the car
+        that should be on this place according to the final arrangement
+        if current empty space position is different from the final order empty
+        space position
+
+        Otherwise picks up the random not arranged car and places it to the empty space
+        '''
+
+        if self.final_car_order[self.empty_spot_position] == 0:
+            car_to_move_position = self.get_not_arranged_car_index()
+            car_to_move = self.current_car_order[car_to_move_position]
+        else:
+            car_to_move = self.final_car_order[self.empty_spot_position]
+            car_to_move_position = self.car_position_dict[car_to_move]
+            self.min_num_steps_left -= 1
+
+        self.print_one_step(car_to_move_position)
+        self.move_car_to_empty_slot(car_to_move_position)
+
+    def reset_variables_to_init_state(self):
+
+        '''Resets variables to the same state when class was initialized
+        Needed for running rearrange_all_cars function again
+        '''
+
+        self.current_car_order = self.initial_car_order
+        self.empty_spot_position = self.current_car_order.index(0)
+        self.car_position_dict = {car:pos for pos, car in \
+                                  enumerate(self.initial_car_order) if car != 0}
+
+    def get_min_num_moves(self):
+
+        '''Returns the minumum number of moves
+        to make final arrangement
+        '''
+        current_final_order_zipped = zip(self.current_car_order, self.final_car_order)
+        num_arranged_cars = len([c1 for c1, c2 in current_final_order_zipped \
+                                 if c1 == c2 and c1])
+        min_steps_left = len(self.initial_car_order) - 1 - num_arranged_cars
+        return min_steps_left
+
+    def rearrange_all_cars(self):
+
+        '''Rearranges all cars and prints the moves
+        Note 1: In case there are more than 1000 moves to print
+        logging into the file is done
+        Note 2: Possible to run the function many times
+
+        '''
+
+        if self.initial_car_order == self.final_car_order:
+            assert ValueError, 'Cars are already in the final order, \
+            no moves are required'
+        elif self.current_car_order == self.final_car_order:
+            self.reset_variables_to_init_state()
+
+        self.min_num_steps_left = self.get_min_num_moves()
+
+        if self.min_num_steps_left > 1000:
+            self.create_log_file()
+            self.print_console = False
+
+        while self.min_num_steps_left:
+            self.rearrange_one_car()
+
+class TestRearrangeCars(unittest.TestCase):

```

Comment:
It's a minor thing, but I think you could split the two things and have a file just for testing, It makes the coding file shorter